snap/snapenv/snapenv.go

```diff
@@ -128,9 +128,9 @@ func userEnv(info *snap.Info, home string, opts *dirs.SnapDirOptions) osutil.Env
 	env["SNAP_REAL_HOME"] = home
 
 	if opts.MigratedToExposedHome {
-		env["XDG_DATA_HOME"] = filepath.Join(info.UserDataDir(home, opts), "data")
-		env["XDG_CONFIG_HOME"] = filepath.Join(info.UserDataDir(home, opts), "config")
-		env["XDG_CACHE_HOME"] = filepath.Join(info.UserDataDir(home, opts), "cache")
+		env["XDG_DATA_HOME"] = filepath.Join(info.UserDataDir(home, opts), "xdg-data")
+		env["XDG_CONFIG_HOME"] = filepath.Join(info.UserDataDir(home, opts), "xdg-config")
+		env["XDG_CACHE_HOME"] = filepath.Join(info.UserDataDir(home, opts), "xdg-cache")
```

Comment:
the spread test should also verify these values, by running a shell app in the snap or something